src/test/java/org/example/services/JobRolesServiceTests.java

```diff
@@ -147,6 +149,18 @@ void getJobRoleById_shouldThrowDatabaseConnectionException_whenDaoThrowsDatabase
                 () -> jobRoleService.getJobRoleById(1));
     }
 
+    @Test
+    void addJobRole_shouldThrowSQLException_whenDaoThrowsSqlException()
+            throws SQLException, DatabaseConnectionException{
+        Mockito.when(databaseConnector.getConnection()).thenReturn(conn);
+
+        Mockito.when(jobRoleDao.insertRole(jobRoleRequest,conn))
+                .thenThrow(SQLException.class);
+
+        assertThrows(SQLException.class, () ->
+                jobRoleDao.insertRole(jobRoleRequest,conn));
```

Comment:
this test also tests the dao, as do some below. please update all tests to call the service method instead